### PR TITLE
Fixed managed nonce in eth_sendTransaction

### PR DESCRIPTION
**Pull request by tkstanczak**

### CODE REVIEW

File: src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthModule.cs [hunk 1]
```diff
@@ -214,7 +214,10 @@ public ResultWrapper<byte[]> eth_getStorageAt(Address address, UInt256 positionI
             }
 
             Account account = _stateReader.GetAccount(header.StateRoot, address);
-            return Task.FromResult(ResultWrapper<UInt256?>.Success(account?.Nonce ?? 0));
+            UInt256 nonce = account?.Nonce ?? 0;
+            _logger.Warn($"{address} nonce is {nonce}");
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthModule.cs:218]: Is this from debugging?

File: src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthModule.cs [hunk 2]
```diff
@@ -313,7 +316,7 @@ public ResultWrapper<byte[]> eth_sign(Address addressData, byte[] message)
         public Task<ResultWrapper<Keccak>> eth_sendTransaction(TransactionForRpc transactionForRpc)
         {
             Transaction tx = transactionForRpc.ToTransactionWithDefaults();
-            return SendTx(tx);
+            return SendTx(tx, TxHandlingOptions.ManagedNonce);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthModule.cs:319]: If nonce is non 0, then will it change it?

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthModule.cs (hunk 1 above)
Comment (tkstanczak) [reply src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthModule.cs:218]: yup

File: src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthModule.cs (hunk 2 above)
Comment (tkstanczak) [reply src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthModule.cs:319]: yeah, I was thinking about it, what do you think?

**Pull request merged**